src/replication.c

```diff
@@ -3336,7 +3372,7 @@ void syncWithPrimary(connection *conn) {
          * that will take care about this. */
         err = sendCommand(conn, "PING", NULL);
         if (err) goto write_error;
-        return;
+        goto cleanup;
```

Comment:
I would like us to break down `syncWithPrimary` in the similar way but in a separate PR. So maybe leave these `syncWithPrimary` changes out of this PR?

Reply:
I prefer a separate PR as it whould be bigger change. So I guess I should not change `syncWithPrimary` at all in this PR right?